lib/krane/deprecated_deploy_task.rb

```diff
@@ -597,7 +597,9 @@ def ejson_keys_secret
     end
 
     def statsd_tags
-      %W(namespace:#{@namespace} sha:#{@current_sha} context:#{@context}) | @namespace_tags
+      tags = %W(namespace:#{@namespace} context:#{@context}) | @namespace_tags
+      @current_sha.nil? ? tags : %W(sha:#{@current_sha}) | tags
+      tags
```

Comment:
The ternary operator will return the evaluated expression so no need for the explicit `tags` line